docs/collect-troubleshooting.rst

```diff
@@ -21,3 +21,82 @@ This generally occurs with forms that have a lot of complex :doc:`logic <form-lo
+
+This error points to a form design issue. If you get the error while filling out a form, please talk to the person who asked you to fill that form out.
+
+If you are the person who built the form, you will need to change the form design to fix the issue. To understand and fix this error, start by looking at the fields listed to identify which part of the field is repeated. If the fields start with ``/data``, the issue has to do with :ref:`repeats <troubleshoot-repeated-field-repeats>`. If the fields start with ``instance('<some name>')``, the issue has to do with :ref:`looking up a value in a dataset <troubleshoot-repeated-field-dataset>`.
+
+.. warning::
+
+  The web form preview will not show this error. Instead, it will default to always using the first repeated value. This is usually not the desired behavior and can be more difficult to identify. We recommend using ODK Collect to test forms with repeats.
+
+.. _troubleshoot-repeated-field-repeats:
+
+Field in repeat used as single value
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Consider this error:
+
+.. code-block:: xml
+
+  XPath evaluation: type mismatch
+  This field is repeated:
+
+  /data/repeat[1]/description[1];/data/repeat[2]/description[1]
+
+Look at the repeated fields in the error and notice ``/data/repeat[1]`` and ``/data/repeat[2]``. This means that there is a repeat with name ``repeat`` in the form. There's an attempt to access the ``description`` field inside that repeat and to use it as a single value, for example in a ``calculate``. If there are more than one ``repeat`` instances added, you will get this crash. In your XLSForm, look for a reference to ``${description}`` outside of ``repeat`` to identify the source of the issue.
+
+Here are some possible fixes depending on what you're trying to do:
+
+* If you're trying to do something with the ``description`` of each repeat instance, you likely can move ``${description}`` inside the repeat instead of using it outside the repeat.
+* If you're trying to use the ``description`` of each repeat instance in another repeat, you can use the :func:`indexed-repeat` function with the :func:`position` function in that second repeat.
+* If you're trying to do something with the ``description`` of all of the repeat instances, you could use a function like :func:`join` from outside the repeat.
+
+You can reproduce this error yourself using `this form definition <https://docs.google.com/spreadsheets/d/1GK8CNawKlDiAx9M4y3df_7gtMQPHshdQ9Bc0fUENzU8/edit?gid=1068911091#gid=1068911091>`__. Start by selecting ``Blue`` to skip the first part of the form. When you get to the repeat, start by adding a single repeat instance and notice that the form works as expected. Then, go back and add another repeat instances, navigate forward, and see this crash.
+
+.. tip::
+
+  Whenever you have a form with a repeat, make sure to test it with 0, 1 and multiple repeat instances. This can help you catch this kind of issue before you send it out to be used.
```

Comment:
Make sure to test it with "0,1, and 2" It's easier to parse. Also oxford comma.

Drop "Before you send it out to be used"